src/components/Offers/New/CreateOfferUtils.js

```diff
@@ -0,0 +1,67 @@
+import { ValidationReasons, validationRulesGenerator } from "../../../utils";
+import { DAY_IN_MS, MONTH_IN_MS, OFFER_MAX_LIFETIME_MONTHS } from "../../../utils/TimeUtils";
+import { MAX_FIELDS, MIN_FIELDS } from "../../utils/offers/FieldOptions";
+import { MAX_TECHNOLOGIES, MIN_TECHNOLOGIES } from "../../utils/offers/TechOptions";
+
+
+export const CreateOfferConstants = {
+    title: {
+        maxLength: 90,
+    },
+    description: {
+        maxLength: 1500,
+    },
+    technologies: {
+        minLength: MIN_TECHNOLOGIES,
+        maxLength: MAX_TECHNOLOGIES,
+    },
+    fields: {
+        minLength: MIN_FIELDS,
+        maxLength: MAX_FIELDS,
+    },
+    contacts: {
+        minLength: 1,
+    },
+    requirements: {
+        minLength: 1,
+    },
+
+};
+
+export const isValidPublishEndDate = (publishDateRaw, publishEndDateRaw) => {
+    const publishDate = Date.parse(publishDateRaw);
+    const publishEndDate = Date.parse(publishEndDateRaw);
+    return publishDate < publishEndDate && (publishEndDate - publishDate) / MONTH_IN_MS <= OFFER_MAX_LIFETIME_MONTHS;
+};
+
+export const defaultDates = {
+    getPublishDate: () => new Date(),
+    getPublishEndDate: () => new Date(Date.parse(new Date()) + (OFFER_MAX_LIFETIME_MONTHS * MONTH_IN_MS) - DAY_IN_MS),
+};
+
+export const generateValidationRule = validationRulesGenerator(CreateOfferConstants);
+
+const HumanReadableErrors = Object.freeze({
+    "no-company-found-with-id": (id) => `No company found with given ID: ${id}.`,
+    "max-concurrent-offers-reached": (val) => `Maximum concurrent offers exceeded. You cannot have more than ${val} offers at a time.`,
+    "must-be-int": () => "Must be an integer.",
+    "company-blocked": () => "Company is blocked. Please contact the team to review this situation if you think this is a mistake.",
+    "company-disabled": () => "Company is disabled. Please enable it or contact the team for help.",
+    "must-be-ISO8601-date": () => ValidationReasons.DATE,
```

Comment:
Do not use Validation Reasons in Human errors directly. Those are not human readable by default

Reply:
Ok, just noticed it below, let's rename the variable, so it is clear it's ok for humans to read.